app/src/androidTest/java/br/com/catbag/gifreduxsample/ui/giflist/GifListActivityTest.java

```diff
@@ -214,9 +227,29 @@ public void whenGifDownloadFailTest() {
     }
 
     @Test
+    public void whenGifListIsEmpty() {
+        mHelper.dispatchFakeAppState(ImmutableAppState.builder().build());
+
+        mActivityTestRule.launchActivity(new Intent());
+
+        onView(withId(R.id.loading))
+                .check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)));
```

Comment:
If the AppState don't change, yes. But without mocks the activity calls the load action, and the Database populates with default gifs...if no previus AppState was saved.